fiftyone/utils/sam.py

```
@@ -647,270 +662,317 @@ class SegmentAnythingModel(fout.TorchSamplesMixin, fout.TorchImageModel):
-                return "boxes"
+        Args:
+            field_mapping (None): a dict mapping required keys to sample fields
 
-            if isinstance(value, fol.Keypoints):
-                return "points"
+        Returns:
+            a :class:`SegmentAnythingImageGetItem` instance
+        """
+        get_item_cls = self.config.get_item_cls
+        if get_item_cls is None:
+            raise Exception("GetItem class is set to None")
```

Ruff (0.15.7)

[warning] Create your own exception

(TRY002)